src/main/java/org/kohsuke/github/GHProject.java

```diff
@@ -0,0 +1,189 @@
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ */
+package org.kohsuke.github;
+
+import java.io.FileNotFoundException;
+import java.io.IOException;
+import java.net.URL;
+import java.util.Locale;
+
+import static org.kohsuke.github.Previews.INERTIA;
+
+/**
+ * A GitHub project.
+ * @see https://developer.github.com/v3/projects/
+ * @author Martin van Zijl
+ */
+public class GHProject extends GHObject {
+    protected GitHub root;
+    protected GHObject owner;
+
+    private String owner_url;
+    private String html_url;
+    private String node_id;
+    private String name;
+    private String body;
+    private int number;
+    private String state;
+    private GHUser creator;
+
+    @Override
+    public URL getHtmlUrl() throws IOException {
+        return GitHub.parseURL(html_url);
+    }
+
+    public GitHub getRoot() {
+        return root;
+    }
+
+    public GHObject getOwner() throws IOException {
+        if(owner == null) {
+            try {
+                if(owner_url.contains("/orgs/")) {
+                    owner = root.retrieve().to(getOwnerUrl().getPath(), GHOrganization.class).wrapUp(root);
+                } else if(owner_url.contains("/users/")) {
+                    owner = root.retrieve().to(getOwnerUrl().getPath(), GHUser.class).wrapUp(root);
+                } else if(owner_url.contains("/repos/")) {
+                    owner = root.retrieve().to(getOwnerUrl().getPath(), GHRepository.class).wrap(root);
+                }
+            } catch (FileNotFoundException e) {
+                return null;
+            }
+        }
+        return owner;
+    }
+
+    public URL getOwnerUrl() {
+        return GitHub.parseURL(owner_url);
+    }
+
+    public String getNode_id() {
+        return node_id;
+    }
+
+    public String getName() {
+        return name;
+    }
+
+    public String getBody() {
+        return body;
+    }
+
+    public int getNumber() {
+        return number;
+    }
+
+    public ProjectState getState() {
+        return Enum.valueOf(ProjectState.class, state.toUpperCase(Locale.ENGLISH));
+    }
+
+    public GHUser getCreator() {
+        return creator;
+    }
+
+    public GHProject wrap(GHRepository repo) {
+        this.owner = repo;
+        this.root = repo.root;
+        return this;
+    }
+
+    public GHProject wrap(GitHub root) {
+        this.root = root;
+        return this;
+    }
+
+    private void edit(String key, Object value) throws IOException {
```

Comment:
@gskjold 

What do you think of the builder/updater style instead? 

https://github.com/github-api/github-api/blob/master/src/main/java/org/kohsuke/github/GHRelease.java
https://github.com/github-api/github-api/blob/master/src/main/java/org/kohsuke/github/GHReleaseBuilder.java
https://github.com/github-api/github-api/blob/master/src/main/java/org/kohsuke/github/GHReleaseUpdater.java

It would reduce round trips but would add more code.  

I don't have a strong opinion.  The project doesn't seem to have one particular style it uses, I'd like to move toward a more consistent style.  I'm wondering what you think. 

(Maybe @kohsuke can comment as well if he's around.)